features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpTelemetryAdapter.java

```diff
@@ -99,40 +100,46 @@ public BmpTelemetryAdapter(final AdapterDefinition adapterConfig,
-        builder.withStringAttribute(peerResource, "bmp", "id", id);
-
-        final BsonDocument stats = document.getDocument("stats");
-        for (final String key : stats.keySet()) {
-            final BsonDocument metric = stats.getDocument(key);
-
-            final String identifier = String.format("bmp_%s_%s", peerAddress, key);
-
-            getInt64(metric, "counter").ifPresent(counter -> {
-                Optional.ofNullable(METRIC_ATTRIBUTE_MAP.get(key)).ifPresent(name -> {
-                    builder.withIdentifiedNumericAttribute(peerResource, "bmp", name, counter, AttributeType.COUNTER, identifier);
-                });
-            });
-            getInt64(metric, "gauge").ifPresent(gauge -> {
-                Optional.ofNullable(METRIC_ATTRIBUTE_MAP.get(key)).ifPresent(name -> {
-                    builder.withIdentifiedNumericAttribute(peerResource, "bmp", name, gauge, AttributeType.GAUGE, identifier);
-                });
-            });
-        }
+        builder.withStringAttribute(peerResource, "bmp", "as", Long.toString(stats.getPeer().getAs()));
+        builder.withStringAttribute(peerResource, "bmp", "id", Long.toString(stats.getPeer().getId()));
+
+        final Function<String, Consumer<Transport.StatisticsReportPacket.Counter>> addCounter = (name) -> (counter) -> {
+            final String identifier = String.format("bmp_%s_%s", peerAddress, name);
+            builder.withIdentifiedNumericAttribute(peerResource, "bmp", name, counter.getCount(), AttributeType.COUNTER, identifier);
+        };
+
+        final Function<String, Consumer<Transport.StatisticsReportPacket.Gauge>> addGauge = (name) -> (gauge) -> {
+            final String identifier = String.format("bmp_%s_%s", peerAddress, name);
+            builder.withIdentifiedNumericAttribute(peerResource, "bmp", name, gauge.getValue(), AttributeType.COUNTER, identifier);
+        };
+
+        Optional.ofNullable(stats.getRejected()).ifPresent(addCounter.apply("rejected"));
+        Optional.ofNullable(stats.getDuplicatePrefix()).ifPresent(addCounter.apply("duplicate_prefix"));
+        Optional.ofNullable(stats.getDuplicateWithdraw()).ifPresent(addCounter.apply("duplicate_withdraw"));
+        Optional.ofNullable(stats.getInvalidUpdateDueToAsConfedLoop()).ifPresent(addCounter.apply("inv_as_confed_loop"));
+        Optional.ofNullable(stats.getInvalidUpdateDueToAsPathLoop()).ifPresent(addCounter.apply("inv_as_path_loop"));
+        Optional.ofNullable(stats.getInvalidUpdateDueToClusterListLoop()).ifPresent(addCounter.apply("inv_cl_loop"));
+        Optional.ofNullable(stats.getInvalidUpdateDueToOriginatorId()).ifPresent(addCounter.apply("inv_originator_id"));
+        Optional.ofNullable(stats.getAdjRibIn()).ifPresent(addGauge.apply("adj_rib_in"));
+        Optional.ofNullable(stats.getAdjRibOut()).ifPresent(addGauge.apply("adj_rib_out"));
+
+        // TODO fooker: Add per AFI counters (perAfiAdjRibIn and perAfiLocRib)
```

Comment:
Worth creating a JIRA for this?

Reply:
Tracking issue is https://issues.opennms.org/browse/NMS-12553